### PR TITLE
Fix borg-list --list-format

### LINKED ISSUE #1486: crash: logging not yet set up in preprocess_args
current git master:
```
$ borg list --list-format '{formatkeys}{LF}' repo::1
Traceback (most recent call last):
  File "/home/tw/w/borg-env/bin/borg", line 9, in <module>
    load_entry_point('borgbackup==1.1.dev333-ngd5edaa5', 'console_scripts', 'borg')()
  File "/home/tw/w/borg/src/borg/archiver.py", line 2359, in main
    args = archiver.get_args(sys.argv, os.environ.get('SSH_ORIGINAL_COMMAND'))
  File "/home/tw/w/borg/src/borg/archiver.py", line 2243, in get_args
    result = self.parse_args(argv[1:])
  File "/home/tw/w/borg/src/borg/archiver.py", line 2259, in parse_args
    args = self.preprocess_args(args)
  File "/home/tw/w/borg/src/borg/archiver.py", line 1254, in preprocess_args
    self.print_warning(warning)
  File "/home/tw/w/borg/src/borg/archiver.py", line 127, in print_warning
    logger.warning(msg)
  File "/home/tw/w/borg/src/borg/logger.py", line 175, in warning
    return self.__logger.warning(*args, **kw)
  File "/home/tw/w/borg/src/borg/logger.py", line 150, in __logger
    raise Exception("tried to call a logger before setup_logging() was called")
Exception: tried to call a logger before setup_logging() was called
```

### PR DESCRIPTION
**Pull request by enkore**
Fixes #1486

### CODE REVIEW

File: src/borg/archiver.py [hunk 1]
```diff
@@ -793,9 +793,9 @@ def write(bytestring):
                 archive = Archive(repository, key, manifest, args.location.archive, cache=cache,
                                   consider_part_files=args.consider_part_files)
 
-                if args.format:
+                if args.format is not None:
                     format = args.format
-                elif args.short:
+                elif args.short is not None:
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py]: isn't that a boolean?

File: src/borg/archiver.py [hunk 2]
```diff
@@ -793,9 +793,9 @@ def write(bytestring):
                 archive = Archive(repository, key, manifest, args.location.archive, cache=cache,
                                   consider_part_files=args.consider_part_files)
 
-                if args.format:
+                if args.format is not None:
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py]: ok

File: src/borg/archiver.py [hunk 3]
```diff
@@ -1247,7 +1247,7 @@ def preprocess_args(self, args):
             for old_name, new_name, warning in deprecations:
                 if arg.startswith(old_name):
                     args[i] = arg.replace(old_name, new_name)
-                    self.print_warning(warning)
+                    print(warning, file=sys.stderr)
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py]: ok

File: src/borg/archiver.py (hunk 1 above)
Comment (enkore) [inline src/borg/archiver.py]: Oops

**Comment by ThomasWaldmann** [comment]
lgtm.

**Pull request merged**